Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/interpolated_corrected_curvatures.h

```diff
@@ -0,0 +1,1649 @@
+    std::queue<Face_descriptor> bfs_queue;
+    std::unordered_set<Face_descriptor> bfs_visited;
+
+    Point_3 vp = get(vpm, v);
+    Vector_3 c = Vector_3(vp.x(), vp.y(), vp.z());
+
+    Vertex_measures<GT> vertex_measures;
+
+    // add the measures of the faces incident to v (excluding the null (boundary) face)
+    for (Face_descriptor f : faces_around_target(halfedge(v, pmesh), pmesh)) {
+      if (f != boost::graph_traits<PolygonMesh>::null_face())
+      {
+        bfs_queue.push(f);
+        bfs_visited.insert(f);
+      }
+    }
+    while (!bfs_queue.empty()) {
+      Face_descriptor fi = bfs_queue.front();
+      bfs_queue.pop();
+
+      // looping over vertices in face to get point coordinates
+      std::vector<Vector_3> x;
+      for (Vertex_descriptor vi : vertices_around_face(halfedge(fi, pmesh), pmesh))
+      {
+        Point_3 pi = get(vpm, vi);
+        x.push_back(Vector_3(pi.x(), pi.y(), pi.z()));
+      }
+
+      // compute the inclusion ratio of the face in the ball
+      const FT f_ratio = face_in_ball_ratio<GT>(x, ball_radius, c);
+
+      // if the face is inside the ball, add the measures
+      // only add the measures for the selected curvatures (area measure is always added)
+      if (f_ratio != 0.0)
+      {
+        vertex_measures.area_measure += f_ratio * get(mu0_map, fi);
+
+        if (is_mean_curvature_selected)
+          vertex_measures.mean_curvature_measure += f_ratio * get(mu1_map, fi);
+
+        if (is_Gaussian_curvature_selected)
+          vertex_measures.gaussian_curvature_measure += f_ratio * get(mu2_map, fi);
+
+        if (is_principal_curvatures_and_directions_selected)
+        {
+          const std::array<FT, 3 * 3> face_anisotropic_measure = get(muXY_map, fi);
+          for (std::size_t i = 0; i < 3 * 3; i++)
+            vertex_measures.anisotropic_measure[i] += f_ratio * face_anisotropic_measure[i];
+        }
+
+        for (Face_descriptor fj : faces_around_face(halfedge(fi, pmesh), pmesh))
+        {
+          if (bfs_visited.find(fj) == bfs_visited.end() && fj != boost::graph_traits<PolygonMesh>::null_face())
+          {
+            bfs_queue.push(fj);
+            bfs_visited.insert(fj);
+          }
+        }
+      }
+    }
+    return vertex_measures;
+  }
+
+  void compute_selected_curvatures() {
+    interpolated_corrected_selected_measures_all_faces();
+
+    for (Vertex_descriptor v : vertices(pmesh))
+    {
+      // expand the computed measures (on faces) to the vertices
+      Vertex_measures<GT> vertex_measures = (ball_radius < 0) ?
+        expand_interpolated_corrected_measure_vertex_no_radius(v) :
+        expand_interpolated_corrected_measure_vertex(v);
+
+      // compute the selected curvatures from the expanded measures and store them in the property maps
+      // if the area measure is zero, the curvature is set to zero
+      if (is_mean_curvature_selected) {
+        vertex_measures.area_measure != 0 ?
```

Comment:
In case this can be a number type other than float or double, CGAL has functions like `is_zero()`, `is_negative()` etc